src/pages/ServiceDetails/ServiceInfo/ServiceInfoVirtualServices.tsx

```diff
@@ -1,84 +1,39 @@
 import * as React from 'react';
-import { Col, Row, Table } from 'patternfly-react';
-import * as resolve from 'table-resolver';
+import {
+  Card,
+  CardBody,
+  EmptyState,
+  EmptyStateBody,
+  EmptyStateVariant,
+  EmptyStateIcon,
+  Grid,
+  GridItem,
+  Title
+} from '@patternfly/react-core';
+import { ICell, IRow, Table, TableHeader, TableBody, TableVariant, cellWidth } from '@patternfly/react-table';
+import { CodeBranchIcon } from '@patternfly/react-icons';
 import { Link } from 'react-router-dom';
 import { ObjectValidation, VirtualService } from '../../../types/IstioObjects';
 import './ServiceInfoVirtualServices.css';
 import LocalTime from '../../../components/Time/LocalTime';
 import { ValidationSummary } from '../../../components/Validations/ValidationSummary';
+import { ServiceDetailsInfo } from '../../../types/ServiceInfo';
 
 interface ServiceInfoVirtualServicesProps {
   virtualServices?: VirtualService[];
+  service: ServiceDetailsInfo;
   validations: { [key: string]: ObjectValidation };
 }
 
 class ServiceInfoVirtualServices extends React.Component<ServiceInfoVirtualServicesProps> {
-  headerFormat = (label, { column }) => <Table.Heading className={column.property}>{label}</Table.Heading>;
-  cellFormat = (value, { column }) => {
-    const props = column.cell.props;
-    const className = props ? props.align : '';
-
-    return <Table.Cell className={className}>{value}</Table.Cell>;
-  };
-
-  columns() {
-    return {
-      columns: [
-        {
-          property: 'status',
-          header: {
-            label: 'Status',
-            formatters: [this.headerFormat]
-          },
-          cell: {
-            formatters: [this.cellFormat],
-            props: {
-              align: 'text-center'
-            }
-          }
-        },
-        {
-          property: 'name',
-          header: {
-            label: 'Name',
-            formatters: [this.headerFormat]
-          },
-          cell: {
-            formatters: [this.cellFormat]
-          }
-        },
-        {
-          property: 'createdAt',
-          header: {
-            label: 'Created at',
-            formatters: [this.headerFormat]
-          },
-          cell: {
-            formatters: [this.cellFormat]
-          }
-        },
-        {
-          property: 'resourceVersion',
-          header: {
-            label: 'Resource version',
-            formatters: [this.headerFormat]
-          },
-          cell: {
-            formatters: [this.cellFormat]
-          }
-        },
-        {
-          property: 'actions',
-          header: {
-            label: 'Actions',
-            formatters: [this.headerFormat]
-          },
-          cell: {
-            formatters: [this.cellFormat]
-          }
-        }
-      ]
-    };
+  columns(): ICell[] {
+    return [
+      { title: 'Status', transforms: [cellWidth(10)] },
```

Comment:
Just a thought. But this is a total improvement from PF4. One line per column instead of 60. :clap: